samples/outreach/blogs/serving_blogpost/temperature_client.py

```diff
@@ -0,0 +1,191 @@
+                            'Number of data points in a day (24 hrs) default: '
+                            'sample rate = 1/10min = 144 data points/day')
+tf.app.flags.DEFINE_integer('num_days',
+                            10,
+                            'Number of days involved in lookback')
+tf.app.flags.DEFINE_integer('steps',
+                            6,
+                            'Sample from datastream at this number of steps. '
+                            'Default: sample rate = 1/10min = 6 steps '
+                            'to sample at 1/hr')
+FLAGS = tf.app.flags.FLAGS
+
+class _ResultMonitor(object):
+  """Monitor for the prediction results."""
+
+  def __init__(self, num_tests, concurrency):
+    self._num_tests = num_tests
+    self._concurrency = concurrency
+    self._mae = 0
+
+    self._done = 0
+    self._active = 0
+    self._condition = threading.Condition()
+
+  def update_mae(self, preds, targets):
+    with self._condition:
+      self._mae += np.mean(np.abs(preds - targets))
+
+  def inc_done(self):
+    with self._condition:
+      self._done += 1
+      self._condition.notify()
+
+  def dec_active(self):
+    with self._condition:
+      self._active -= 1
+      self._condition.notify()
+
+  def get_mae(self):
+    with self._condition:
+      while self._done != self._num_tests:
+        self._condition.wait()
+      return self._mae / float(self._num_tests)
+
+  def throttle(self):
+    with self._condition:
+      while self._active == self._concurrency:
+        self._condition.wait()
+      self._active += 1
+
+
+def _create_rpc_callback(label, result_counter):
+  """Creates RPC callback function.
+
+  Args:
+    label: The correct label for the predicted example.
+    result_counter: Counter for the prediction result.
+  Returns:
+    The callback function.
+  """
+  def _callback(result_future):
+    """Callback function.
+
+    Calculates the statistics for the prediction result.
+
+    Args:
+      result_future: Result future of the RPC.
+    """
+    exception = result_future.exception()
+    if exception:
+      print("An error occured: {}".format(exception))
+    else:
+      sys.stdout.write('.')
+      sys.stdout.flush()
+      prediction = np.array(
+          result_future.result().outputs['temperature'].float_val)
+      result_counter.update_mae(prediction, label)
+    result_counter.inc_done()
+    result_counter.dec_active()
+  return _callback
+
+
+def do_inference(hostport, concurrency, num_tests):
+  """Tests PredictionService with concurrent requests.
+
+  Args:
+    hostport: Host:port address of the PredictionService.
+    work_dir: The full path of working directory for test data set.
+    concurrency: Maximum number of concurrent requests.
+    num_tests: Number of test images to use.
+
+  Returns:
+    The classification error rate.
+
+  Raises:
+    IOError: An error occurred processing test data set.
+  """
+
+  ds = data_utils.Dataset(FLAGS.data_path,
```

Comment:
Shouldn't this be fine/depend on the use case? The model expects data that is standard scaled (zero meaned, unit std). When we perform inference on a new file, we recalculate the normalization coeffs to ensure this, since the distribuiton of data may have changed.

We could build this into the model and freeze the constants associated with it, but if the distribution of data changes, the model may not be as performant since the data isn't what it expects

Reply:
>The model expects data that is standard scaled (zero meaned, unit std)

zero-meaned, unit std over the whole domain.
But run a file of summer data (average high: sin(x)+y).
Then a file of winter data (average low: sin(x)-y).

The model can't see the difference. It will output the exact same values for both. That doesn't seem right.

Reply:
Updated to not normalize when performing inference